glowing_waffles/visualization/aij_plots.py

```diff
@@ -0,0 +1,101 @@
+    ----------
+    raw_radius : array
+        the distance of each pixel from the object of interest
+
+    raw_counts : array
+        the counts of each pixel
+
+    binned_radius : array
+        pixels grouped by distance from object of interest
+
+    binned_counts : array
+        average counts of each group of pixels
+
+    HWHM : number
+        half width half max, 1/2 * FWHM
+
+    plot_title : optional, string
+        title of plot
+
+    file_name : optional, string
+        if entered, file will save as png with this name
+    gap : number
+        the distance between the aperture and the inner annulus
+    annulus_width : number
+        the distance between the inner and outer annulus
+    """
+    radius = HWHM * 4
+    plt.figure(figsize=(20,10))
+    plt.grid(True)
+    inner_annulus = radius + gap
+    outer_annulus = radius + annulus_width
+
+
+    # plot the raw radius and raw counts
+    plt.plot(raw_radius, raw_counts, linestyle='none', marker="s", markerfacecolor='none', color='blue')
+
+    # plot the binned radius and binned counts
+    plt.plot(binned_radius, binned_counts, color = 'magenta',linewidth = '4.0')
+
+    # draw vertical line at HWHM and label it
+    plt.vlines(HWHM,-0.2,1.2, linestyle = (0, (5, 10)), color = '#00cc00')
+    plt.annotate(f"HWHM {HWHM}", (HWHM - 1 ,-0.25), fontsize = 15, color = '#00cc00')
+    
+    # label axis
+    plt.xlabel('Radius (pixels)', fontsize = 20)
+    plt.ylabel('ADU', fontsize = 20)
+
+    # draw vertical line at the radius and label it
+    plt.vlines(radius,-0.2, binned_counts[0], color = 'red')
+    plt.annotate(f"Radius {radius}", (radius - 1 ,-0.25), fontsize = 15, color = 'red')
+    plt.hlines(binned_counts[0], binned_counts[0], radius, color = 'red')
+
+    # label the source
+    plt.annotate('SOURCE', (radius - 2, binned_counts[0] + 0.02), fontsize = 15, color = 'red')
+
+    #draw vertical lines at the background and label it
+    plt.vlines(inner_annulus, -0.2, binned_counts[0], color = 'red')
+    plt.vlines(outer_annulus, -0.2, binned_counts[0], color = 'red')
+    plt.hlines(binned_counts[0], inner_annulus, outer_annulus, color = 'red')
+    plt.annotate('BACKGROUND', (inner_annulus, binned_counts[0] + 0.02), fontsize = 15, color = 'red')
+    plt.annotate(f"Back> {inner_annulus}", (inner_annulus-1, -0.25), fontsize = 15, color = 'red')
+    plt.annotate(f"<Back {outer_annulus}", (outer_annulus-1, -0.25), fontsize = 15, color = 'red')
+
+    #title the plot
+    title_string = [f"{plot_title}", f"FWHM:{HWHM*2} pixels"]
+    plt.title('\n'.join(title_string))
+
+    #save plot as png
+    if file_name:
+        plt.savefig(f"{file_name}.png")
+
+def predict_ingress_egress(ingress_time, egress_time):
+    """
+    Parameters
+    ----------
+    ingress_time : number
+        the beginning of an exoplanet transit
+
+    egress_time : number
+        the end of an exoplanet transit
+    """
+    ymin,ymax = plt.ylim()
+    end_line = ymin - 0.08
```

Comment:
If the 0.08 might need to be adjusted for each plot should it be a (maybe optional) function argument?

I assume the other offsets below (0.009 and 0.1) are similar?